### PR TITLE
[ONNX][#8838] QLinearSigmoid contrib op and Bug Fix for DequantizeLinear

### PR DESCRIPTION
**Pull request by arangasa**
This PR implements com.microsoft.QLinearSigmoid (tracked here). This is implemented as dequantize->sigmoid->quantize for now.

This also fixes an issue with DequantizeLinear for tensors with rank=1. The current default axis of 1 for _impl_v13 leads to failures with rank=1 tensors. A new test is added for an existing op (QLinearMul) for this case.

### CODE REVIEW

**Comment by tmoreau89** [comment]
CC - @anwang2009 @mbrookhart @AndrewZhaoLuo

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -3176,7 +3176,7 @@ def _impl_v10(cls, inputs, attr, params):
     @classmethod
     def _impl_v13(cls, inputs, attr, params):
         data, scale, zp = inputs
-        axis = attr.get("axis", 1)
+        axis = attr.get("axis", 0)
```
Comment (mbrookhart) [inline python/tvm/relay/frontend/onnx.py:3179]: I don't think this is correct. The default in the ONNX spec is 1. I think we should leave it as 1, I think there will be cases that fail with this change, but have code to check for data with 0D or 1D shapes and set it to 0 in that case?

https://github.com/onnx/onnx/blob/master/docs/Changelog.md#DequantizeLinear-13

**Review by mbrookhart — CHANGES_REQUESTED** [review]
I think the bugfix with DequantizeLinear isn't quite correct, but the QLinearSigmoid stuff looks great! Thank you.

File: python/tvm/relay/frontend/onnx.py [hunk 2]
```diff
@@ -3351,6 +3351,28 @@ def _impl_v10(cls, inputs, attr, params):
...
+        y_scale = fold_constant(get_scalar(inputs[3], params))
+        y_zero_point = get_scalar(inputs[4], params, "int32")
+
+        dtype = infer_type(x).checked_type.dtype
+
+        ## Apparently, onnxruntime doesn't do this op in integer, they dequantize to fp32
+        ## and then requantize after:
+        ## https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/
+        ## providers/dml/DmlExecutionProvider/src/GraphTransformer.cpp#L245
+        x = _qnn.op.dequantize(inputs[0], x_scale, x_zero_point)
```
Comment (anwang2009) [inline python/tvm/relay/frontend/onnx.py:3371]: nit
```suggestion
        x = _qnn.op.dequantize(x, x_scale, x_zero_point)
```

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM

**Review by anwang2009 — APPROVED** [review]
lgtm!

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (anwang2009) [reply python/tvm/relay/frontend/onnx.py:3179]: yeah confirmed docs are correct, here is where the default is set: https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cpu/tensor/quantize_linear.h#L17

**Comment by mbrookhart** [comment]
The de-quantize change did cause those tests to fail in CI. I think we need a better solution there, but otherwise, it looks good.

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (masahi) [reply python/tvm/relay/frontend/onnx.py:3179]: I've also hit this issue when I was testing a quantized tflite model, the change below worked (as @mbrookhart suggested)
https://github.com/masahi/tvm/blob/e649837fb4f74abbbb770e0a3f6bf175f407cc7d/python/tvm/relay/frontend/onnx.py#L3199-L3202

**Comment by arangasa** [comment]
> The de-quantize change did cause those tests to fail in CI. I think we need a better solution there, but otherwise, it looks good.

Thank you, @mbrookhart , @anwang2009 , @masahi , @AndrewZhaoLuo for reviewing, catching the issue with this change, and for suggesting a solution. Working on this.

**Review by masahi — APPROVED** [review]
cc @mbrookhart

**Review by mbrookhart — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**